src/domains/Dashboard/TopDataCards/TopDataCards.tsx

```
@@ -0,0 +1,117 @@
+import { Grid } from '@mui/material';
+import { BigTopDataCard } from './components/BigTopDataCard/BigTopDataCard';
+import { SmallTopDataCard } from './components/SmallTopDataCard/SmallTopDataCard';
+import { EarningGraph } from './components/EarningGraph/EarningGraph';
+import { OrderButtons } from './components/OrderButtons/OrderButtons';
+import { TableChart, Storefront, Redeem, LocalAtm } from '@mui/icons-material';
+import { MiddleValueIcon, StyledCircularProgress } from './TopDataCards.styles';
+import { EarningButton } from './components/EarningButton/EarningButton';
+import { useState } from 'react';
+import { useDashboardValues } from './hooks/useDashboardValues';
+import { useTheme } from '@mui/material';
+
+export enum CardType {
+    Order = 'Order',
+    Earning = 'Earning',
+}
+
+export enum DataType {
+    Month = 'Month',
+    Year = 'Year',
+}
+
+enum FetchError {
+    Err = 'Err',
+}
+
+export const TopDataCards = () => {
+    const [dataType, setDataType] = useState(DataType.Month);
+    const { dashboardValues, isLoading, isError, isSuccess } = useDashboardValues(dataType);
+    const theme = useTheme();
+
+    return (
+        <>
+            <Grid item xs={12} sm={6} lg={4}>
+                <BigTopDataCard
+                    cardType={CardType.Earning}
+                    leftTopButton={<LocalAtm />}
+                    rightTopButton={<EarningButton />}
+                    value={
+                        isSuccess ? (
+                            dashboardValues.earningTotal
+                        ) : isLoading ? (
+                            <StyledCircularProgress size={theme.spacing(3.5)} />
+                        ) : isError ? (
+                            FetchError.Err
```

Hmm, in some projects that might be the use case for this FetchError enum. I can see the case like this:
- We want to hide real error from user, but we want to inform that something happend so we use generic error message. But then this enum should be placed in the shared/enum folder so we can re use it anywhere from the app

What is more important, that in our app we already have i18n which gives us reusable strings. So we can add 'common' translations then place there something like 'network error' and call it when we recive fetch error. 

Error at the component itself i think some red icon with '!' could be a good indicator that somethings wrong. And i think that in the future we will implement snackbars for communication with the user.